### PR TITLE
[BULK] Fix 50 broken links in AspNetCore.Docs as of 6/22/2023

### PR DESCRIPTION
**Pull request by v-thepet**
Although the AspNetCore.Docs broken links dashboard https://aka.ms/brokenlinkdashboard showed 21 broken links on 6/19/2023, and they were all fixed or removed in PR #29569 which went live on 6/21, the broken link dashboard as of 6/22/2023 now shows 50 broken links. These are all "new" in that they weren't on the 6/19 report, although are in old content.

See https://github.com/dotnet/AspNetDocs/pull/843 for an analogous situation with AspNetDocs.

And as in that PR, a couple more broken links cropped up in the first build of this current PR that weren't listed on the dashboard reports so far.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/components/layouts.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/blazor/components/layouts.md) | [ASP.NET Core Blazor layouts](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/layouts?branch=pr-en-us-29637) |
| [aspnetcore/blazor/fundamentals/startup.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/blazor/fundamentals/startup.md) | [ASP.NET Core Blazor startup](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?branch=pr-en-us-29637) |
| [aspnetcore/blazor/security/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/blazor/security/index.md) | [ASP.NET Core Blazor authentication and authorization](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/security/index?branch=pr-en-us-29637) |
| [aspnetcore/blazor/security/server/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/blazor/security/server/index.md) | [[Visual Studio](#tab/visual-studio)](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/security/server/index?branch=pr-en-us-29637) |
| [aspnetcore/fundamentals/url-rewriting.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/fundamentals/url-rewriting.md) | [aspnetcore/fundamentals/url-rewriting](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?branch=pr-en-us-29637) |
| [aspnetcore/host-and-deploy/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/host-and-deploy/index.md) | [Host and deploy ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/host-and-deploy/index?branch=pr-en-us-29637) |
| [aspnetcore/performance/response-compression.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/performance/response-compression.md) | [Response compression in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/performance/response-compression?branch=pr-en-us-29637) |
| [aspnetcore/release-notes/aspnetcore-8.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/a8853aef014035e707e726ce6564925d9c56ae87/aspnetcore/release-notes/aspnetcore-8.0.md) | [What's new in ASP.NET Core 8.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?branch=pr-en-us-29637) |

### CODE REVIEW

**Review by guardrex — CHANGES_REQUESTED** [review]
Hello @v-thepet ... A number of the Blazor ones (primarily the `Client`-segmented links) will need to change. The product unit is dropping a template option that uses these resources. They're working on their `main` branch, which is for the upcoming 8.0 release. However, we need to support the old template in our docs. What I'll do is track down the correct 7.0 release files for these links, and we'll peg these at 7.0, which will be the last release for their availability. Stand-by for a bit. I'll get the links to you *ASAP*.

File: aspnetcore/blazor/components/layouts.md [hunk 1]
```diff
@@ -93,8 +93,8 @@ In an app created from a [Blazor project template](xref:blazor/project-structure
...
 
-* [Blazor Server `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Shared/MainLayout.razor.css)
-* [Blazor WebAssembly `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/MainLayout.razor.css)
+* [Blazor Server `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebAssembly/Samples/HostedBlazorWebassemblyApp/Client/Shared/MainLayout.razor.css)
+* [Blazor WebAssembly `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Shared/MainLayout.razor.css)
```
Comment (guardrex) [inline aspnetcore/blazor/components/layouts.md:97]:
```suggestion
* [Blazor WebAssembly `MainLayout.razor.css`](
https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/MainLayout.razor.css)
```

File: aspnetcore/blazor/components/layouts.md [hunk 2]
```diff
@@ -93,8 +93,8 @@ In an app created from a [Blazor project template](xref:blazor/project-structure
 
 [Blazor's CSS isolation feature](xref:blazor/components/css-isolation) applies isolated CSS styles to the `MainLayout` component. By convention, the styles are provided by the accompanying stylesheet of the same name, `Shared/MainLayout.razor.css`. The ASP.NET Core framework implementation of the stylesheet is available for inspection in the ASP.NET Core reference source (`dotnet/aspnetcore` GitHub repository):
 
-* [Blazor Server `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Shared/MainLayout.razor.css)
-* [Blazor WebAssembly `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/MainLayout.razor.css)
+* [Blazor Server `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebAssembly/Samples/HostedBlazorWebassemblyApp/Client/Shared/MainLayout.razor.css)
```
Comment (guardrex) [inline aspnetcore/blazor/components/layouts.md:96]:
```suggestion
* [Blazor Server `MainLayout.razor.css`](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Shared/MainLayout.razor.css)
```

File: aspnetcore/blazor/fundamentals/startup.md [hunk 3]
```diff
@@ -403,8 +403,8 @@ The default round progress indicator is implemented in HTML in the `wwwroot/inde
 
 To review the Blazor WebAssembly project template markup and styling for the default progress indicators, see the ASP.NET Core reference source:
 
-* [`wwwroot/index.html`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/index.html)
-* [`app.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/css/app.css)
+* [`wwwroot/index.html`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/wwwroot/index.html)
```
Comment (guardrex) [inline aspnetcore/blazor/fundamentals/startup.md:406]:
```suggestion
* [`wwwroot/index.html`](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/index.html)
```

File: aspnetcore/blazor/fundamentals/startup.md [hunk 4]
```diff
@@ -403,8 +403,8 @@ The default round progress indicator is implemented in HTML in the `wwwroot/inde
...
+* [`app.css`](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/wwwroot/css/app.css)
```
Comment (guardrex) [inline aspnetcore/blazor/fundamentals/startup.md:407]:
```suggestion
* [`app.css`](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/wwwroot/css/app.css)
```

File: aspnetcore/blazor/security/includes/app-component.md [hunk 5]
```diff
@@ -7,6 +7,6 @@ The `App` component (`App.razor`) is similar to the `App` component found in Bla
 Due to changes in the framework across releases of ASP.NET Core, Razor markup for the `App` component (`App.razor`) isn't shown in this section. To inspect the markup of the component for a given release, use ***either*** of the following approaches:
 
 * Create an app provisioned for authentication from the default Blazor WebAssembly project template for the version of ASP.NET Core that you intend to use. Inspect the `App` component (`App.razor`) in the generated app.
-* Inspect the `App` component (`App.razor`) in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/App.razor).
+* Inspect the `App` component (`App.razor`) in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/App.razor).
```
Comment (guardrex) [inline aspnetcore/blazor/security/includes/app-component.md:10]:
```suggestion
* Inspect the `App` component (`App.razor`) in [reference source](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/App.razor).
```

File: aspnetcore/blazor/security/includes/logindisplay-component.md [hunk 6]
```diff
@@ -11,6 +11,6 @@ The `LoginDisplay` component (`Shared/LoginDisplay.razor`) is rendered in the `M
 Due to changes in the framework across releases of ASP.NET Core, Razor markup for the `LoginDisplay` component isn't shown in this section. To inspect the markup of the component for a given release, use ***either*** of the following approaches:
 
 * Create an app provisioned for authentication from the default Blazor WebAssembly project template for the version of ASP.NET Core that you intend to use. Inspect the `LoginDisplay` component in the generated app.
-* Inspect the `LoginDisplay` component in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/LoginDisplay.IndividualLocalAuth.razor). The templated content for `Hosted` equal to `true` is used.
+* Inspect the `LoginDisplay` component in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Shared/LoginDisplay.IndividualLocalAuth.razor). The templated content for `Hosted` equal to `true` is used.
```
Comment (guardrex) [inline aspnetcore/blazor/security/includes/logindisplay-component.md:14]:
```suggestion
* Inspect the `LoginDisplay` component in [reference source](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/LoginDisplay.IndividualLocalAuth.razor). The templated content for `Hosted` equal to `true` is used.
```

File: aspnetcore/blazor/security/includes/redirecttologin-component.md [hunk 7]
```diff
@@ -5,6 +5,6 @@ The `RedirectToLogin` component (`Shared/RedirectToLogin.razor`):
   * [Navigation history state](xref:blazor/fundamentals/routing#navigation-history-state) in ASP.NET Core 7.0 or later.
   * A query string in ASP.NET Core 6.0 or earlier.
 
-Inspect the `RedirectToLogin` component in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/RedirectToLogin.razor).
+Inspect the `RedirectToLogin` component in [reference source](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Shared/RedirectToLogin.razor).
```
Comment (guardrex) [inline aspnetcore/blazor/security/includes/redirecttologin-component.md:8]:
```suggestion
Inspect the `RedirectToLogin` component in [reference source](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/RedirectToLogin.razor).
```

File: aspnetcore/blazor/security/index.md [hunk 8]
```diff
@@ -505,7 +505,7 @@ If the `<NotAuthorized>` tag isn't specified, the <xref:Microsoft.AspNetCore.Com
...
+An app created from the Blazor WebAssembly project template with authentication enabled includes a [`RedirectToLogin` component (`Shared/RedirectToLogin.razor` reference source)](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Shared/RedirectToLogin.razor), which is positioned in the `<NotAuthorized>` content of the Blazor Router in the template's [`App` component (reference source)](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/App.razor). When a user isn't authenticated (`context.User.Identity?.IsAuthenticated != true`), the `RedirectToLogin` component redirects the browser to the `authentication/login` endpoint for authentication. The user is returned to the requested URL after authenticating with the identity provider.
```
Comment (guardrex) [inline aspnetcore/blazor/security/index.md:508]:
```suggestion
An app created from the Blazor WebAssembly project template with authentication enabled includes a [`RedirectToLogin` component (`Shared/RedirectToLogin.razor` reference source)](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/RedirectToLogin.razor), which is positioned in the `<NotAuthorized>` content of the Blazor Router in the template's [`App` component (reference source)](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/App.razor). When a user isn't authenticated (`context.User.Identity?.IsAuthenticated != true`), the `RedirectToLogin` component redirects the browser to the `authentication/login` endpoint for authentication. The user is returned to the requested URL after authenticating with the identity provider.
```

File: aspnetcore/blazor/security/server/index.md [hunk 9]
```diff
@@ -541,7 +541,7 @@ Two additional abstractions participate in managing authentication state:
...
 
-  For a Blazor Server app created from the Blazor Server project template with authentication enabled, the default 30 minute revalidation interval can be adjusted in [`RevalidatingIdentityAuthenticationStateProvider` (`Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs`)](https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs). The following example shortens the interval to 20 minutes:
+  For a Blazor Server app created from the Blazor Server project template with authentication enabled, the default 30 minute revalidation interval can be adjusted in `RevalidatingIdentityAuthenticationStateProvider` (`Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs`). The following example shortens the interval to 20 minutes:
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:544]:
```suggestion
  For a Blazor Server app created from the Blazor Server project template with authentication enabled, the default 30 minute revalidation interval can be adjusted in [`RevalidatingIdentityAuthenticationStateProvider` (`Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs`)](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs). The following example shortens the interval to 20 minutes:
```

**Review by guardrex — CHANGES_REQUESTED** [review]
Ok ... I think that gets the Blazor ones squared away.

I'll call for Rick now on the rest because he was handling the URL Rewriting ones on a recent PR.

**Comment by v-thepet** [comment]
@guardrex I have implemented your changes. @Rick-Anderson Please make/suggest any additional changes. The aim of this PR is to fix any links that showed broken on the latest 6/22 dashboard, but perhaps we need a new or updated work item if this is to be an ongoing task. Thanks

**Review by guardrex — APPROVED** [review]
Thanks, @v-thepet. I'll approve now for the Blazor updates, and Rick can take a look at the rest.

> if this is to be an ongoing task

We still have two more preview releases to go, so I wouldn't be surprised if more broken links pop up over the next few months.

As I proceed with the .NET 8 work, I'll be checking the Blazor ref source cross-links and dealing with the versioning where needed. For now, feel free to attempt swapping "`release/7.0`" in place of "`main`" for any new broken links.

**Review by Rick-Anderson — APPROVED** [review]

**Pull request merged**